mitodl/micromasters
# Added auto-incrementing `student_id` field to `Profile`

#### relevant tickets:

this accomplishes part of #69

#### what's this pr do?

this adds a new attribute to the Profile model (`student_id`) that we
basically keep incremented.

to do so we override the `save` method on the Profile model, so that we
can insert our own logic to set the correct `student_id` on `self`
before we save (only if the model is new or the `student_id` field is
currently empty).

we also add this attribute to the `ProfileSerializer`, which is what
spits the profile JSON to the frontend.

#### where should the reviewer start?

the migration and the model change:

	new file:   profiles/migrations/0005_profile_student_id.py
	modified:   profiles/models.py

the serializer change is smaller, I think

	modified:   profiles/serializers.py

#### how should this be manually tested?

I think the key thing is testing that when you save a `Profile` instance
it has a unique `student_id`. It should save a `student_id` when saving
both new and previously created objects.

#### any background context?

I modeled this change over how we accomplished a somewhat (I think?)
similar thing over in Lore.

The relevant pieces are here:

https://github.com/mitodl/lore/blob/master/learningresources/models.py#L163

https://github.com/mitodl/lore/blob/master/rest/util.py#L66
